adapters/Mintegral/MintegralAdapter/Bidding/GADMAdapterMintegralRewardedAdRenderer.m

```diff
@@ -0,0 +1,141 @@
+    __block GADMediationRewardedLoadCompletionHandler originalCompletionHandler =
+    [completionHandler copy];
+    _adLoadCompletionHandler = ^id<GADMediationRewardedAdEventDelegate>(
+                                                                        _Nullable id<GADMediationRewardedAd> ad, NSError *_Nullable error) {
+        if (atomic_flag_test_and_set(&completionHandlerCalled)) {
+            return nil;
+        }
+        id<GADMediationRewardedAdEventDelegate> delegate = nil;
+        if (originalCompletionHandler) {
+            delegate = originalCompletionHandler(ad, error);
+        }
+        originalCompletionHandler = nil;
+        return delegate;
+    };
+    
+    NSString *adUnitId = adConfiguration.credentials.settings[GADMAdapterMintegralAdUnitID];
+    NSString *placementId = adConfiguration.credentials.settings[GADMAdapterMintegralPlacementID];
+
+    if ([GADMAdapterMintegralUtils isStringEmpty:adUnitId]) {
+        NSError *error =
+        GADMAdapterMintegralErrorWithCodeAndDescription(GADMintegralErrorInvalidServerParameters, @"Ad Unit ID connot be nil.");
+        _adLoadCompletionHandler(nil, error);
+        return;
+    }
+        
+    _rewardedAd = [MTGBidRewardAdManager sharedInstance];
+    [_rewardedAd loadVideoWithBidToken:_adConfiguration.bidResponse placementId:placementId unitId:adUnitId delegate:self];
+}
+
+#pragma mark MTGRewardAdLoadDelegate
+- (void)onVideoAdLoadSuccess:(nullable NSString *)placementId unitId:(nullable NSString *)unitId {
+    
+    if (_adLoadCompletionHandler) {
+        _adEventDelegate = _adLoadCompletionHandler(self, nil);
+    }
+}
+
+- (void)onVideoAdLoadFailed:(nullable NSString *)placementId unitId:(nullable NSString *)unitId error:(nonnull NSError *)error {
+    
+    if (_adLoadCompletionHandler) {
+        _adLoadCompletionHandler(nil, error);
+    }
+}
+
+#pragma mark MTGRewardAdShowDelegate
+- (void)onVideoAdShowSuccess:(nullable NSString *)placementId unitId:(nullable NSString *)unitId {
+    [_adEventDelegate willPresentFullScreenView];
+    [_adEventDelegate reportImpression];
+    [_adEventDelegate didStartVideo];
+}
+
+- (void)onVideoAdShowFailed:(nullable NSString *)placementId unitId:(nullable NSString *)unitId withError:(nonnull NSError *)error {
+    [_adEventDelegate didFailToPresentWithError:error];
+}
+
+- (void)onVideoPlayCompleted:(nullable NSString *)placementId unitId:(nullable NSString *)unitId{
+    [_adEventDelegate didEndVideo];
+}
+
+- (void)onVideoAdClicked:(nullable NSString *)placementId unitId:(nullable NSString *)unitId {
+    [_adEventDelegate reportClick];
+}
+
+- (void)onVideoAdDismissed:(nullable NSString *)placementId unitId:(nullable NSString *)unitId withConverted:(BOOL)converted withRewardInfo:(nullable MTGRewardAdInfo *)rewardInfo {
+   
+    if (converted) {
+        [_adEventDelegate didRewardUser];
+    }
+}
+
+- (void)onVideoAdDidClosed:(nullable NSString *)placementId unitId:(nullable NSString *)unitId {
```

Comment:
Just to confirm, this will be called after `onVideoAdDismissed` correct?

Reply:
What is the difference between this and `onVideoAdDismissed`?

Reply:
Just to confirm, this will be called after onVideoAdDismissed correct? yes

What is the difference between this and onVideoAdDismissed?
onVideoAdDismissed = will close
onVideoAdDidClosed = did close